flocker/testtools/_flaky.py

```diff
@@ -22,10 +37,179 @@ def flaky(jira_key):
 
     # TODO (see FLOC-3281):
     # - allow specifying which exceptions are expected
-    # - retry on expected exceptions
-    #   - parametrize on retry options
     # - provide interesting & parseable logs for flaky tests
 
+    if isinstance(jira_keys, basestring):
+        jira_keys = [jira_keys]
+
+    annotation = _FlakyAnnotation(
+        jira_keys=pset(jira_keys), max_runs=max_runs, min_passes=min_passes)
+
     def wrapper(test_method):
+        setattr(test_method, _FLAKY_ATTRIBUTE, annotation)
```

Comment:
:grey_question: Do you want to do anything special if someone added the flaky decorator twice? looks like this would just stomp on the inner one presently. That's probably okay, although we might lose some jira_keys. Okay as is  / at least we won't be retrying a retry loop (by my reading anyway)